src/fit.jl

```diff
@@ -165,26 +184,65 @@ function eval_metrics!(results::Dict{String,Dict{String,Vector{Float64}}},
 end
 
 
-function store_scores!(results::Dict{String,Dict{String,Vector{Float64}}},
-                       estimator::LGBMEstimator, iter::Integer, evalname::String,
-                       scores::Vector{Cdouble}, metrics::Vector{String})
+function store_scores!(
+    results::Dict{String,Dict{String,Vector{Float64}}},
+    estimator::LGBMEstimator,
+    store_idx::Integer,
+    num_evals::Integer,
+    evalname::String,
+    scores::Vector{Cdouble},
+    metrics::Vector{String},
+)
+
     for (metric_idx, metric_name) in enumerate(metrics)
         if !haskey(results, evalname)
-            num_evals = cld(estimator.num_iterations, estimator.metric_freq)
             results[evalname] = Dict{String,Vector{Float64}}()
             results[evalname][metric_name] = Array{Float64}(undef,num_evals)
         elseif !haskey(results[evalname], metric_name)
-            num_evals = cld(estimator.num_iterations, estimator.metric_freq)
             results[evalname][metric_name] = Array{Float64}(undef,num_evals)
         end
-        eval_idx = cld(iter, estimator.metric_freq)
-        results[evalname][metric_name][eval_idx] = scores[metric_idx]
+        results[evalname][metric_name][store_idx] = scores[metric_idx]
     end
 
     return nothing
 end
 
 
+function merge_scores(
```

Comment:
This is the functionality for merging metrics histories if training is completed in multiple batches